### PR TITLE
endecode_test.c: Avoid running the SM2 tests with 3.0.0 FIPS provider

### LINKED ISSUE #25326: Provider compatibility test is broken with 3.0.0 fips provider
As a result of merging https://github.com/openssl/openssl/pull/25266 the provider compatibility test is broken. The newly added testcases for some reason do not work with the 3.0.0 fips provider but they work with any other newer version.

### PR DESCRIPTION
**Pull request by t8m**
Fixes #25326

### CODE REVIEW

**Review by levitte — APPROVED** [review]
Agree urgent

**Review by nhorman — APPROVED** [review]

**Comment by t8m** [comment]
Merged to all the active branches. Thank you for the reviews.

File: test/endecode_test.c [hunk 1]
```diff
@@ -1451,7 +1451,10 @@ int setup_tests(void)
         ADD_TEST_SUITE_LEGACY(ECExplicitTri2G);
 # endif
 # ifndef OPENSSL_NO_SM2
-        ADD_TEST_SUITE(SM2);
+        if (!is_fips_3_0_0) {
```
Comment (paulidale) [inline test/endecode_test.c:1454]: This isn't adequate.  is_fips_3_0_0 will be true when the FIPS provider isn't present.

Comment (tom-cosgrove-arm) [reply test/endecode_test.c:1454]: Could there be a comment where this variable is declared to explain what its potential values mean? And: is there a better name for this? Even just `fips_3_0_0_status` (and some `#define`s) is likely to at least cause people to tread carefully - a variable with just "is-foo" is likely to have its semantics assumed

**Pull request closed without merging**